test/net/sourceforge/kolmafia/textui/ParserTest.java

```diff
@@ -1318,21 +1580,42 @@ public void clearRevision() {
         valid(
             "proper remove",
             "int[] map; remove map[0];",
-            Arrays.asList("int", "[", "]", "map", ";", "remove", "map", "[", "0", "]", ";")));
+            Arrays.asList("int", "[", "]", "map", ";", "remove", "map", "[", "0", "]", ";"),
+            Arrays.asList(
+                "1-1", "1-4", "1-5", "1-7", "1-10", "1-12", "1-19", "1-22", "1-23", "1-24",
+                "1-25")));
   }
 
   @ParameterizedTest
   @MethodSource("data")
   public void testScriptValidity(ScriptData script) {
     if (script instanceof InvalidScriptData) {
-      ScriptException e = assertThrows(ScriptException.class, script.parser::parse, script.desc);
-      assertThat(
-          script.desc, e.getMessage(), containsString(((InvalidScriptData) script).errorText));
+      testInvalidScript((InvalidScriptData) script);
       return;
     }
 
+    testValidScript((ValidScriptData) script);
+  }
+
+  private static void testInvalidScript(final InvalidScriptData script) {
```

Comment:
I was suggesting splitting the parameterized data into "invalid cases" and "valid cases" but I guess this works too. Either way if we want to do that in a future PR, we can.